apps/src/code-studio/components/ExportAllowedDialog.jsx

```diff
@@ -0,0 +1,743 @@
+        exportError: null,
+        expoUri,
+        expoSnackId,
+        iconUri,
+        splashImageUri
+      });
+    } catch (e) {
+      this.setState({
+        exporting: false,
+        exportProjectUpdatedAt: null,
+        expoUri: null,
+        expoSnackId: null,
+        exportError:
+          'Failed to publish project to Expo. Please try again later.'
+      });
+    }
+  }
+
+  async publishAndGenerateApk() {
+    if (this.state.apkUri) {
+      // We have already have generated an APK
+      return;
+    }
+
+    await this.publishExpoExport();
+
+    const {exportApp} = this.props;
+    const {expoSnackId, iconUri, splashImageUri} = this.state;
+
+    if (!expoSnackId) {
+      // We failed to generate a snackId
+      return;
+    }
+
+    this.setState({generatingApk: true});
+    try {
+      const apkUri = await exportApp({
+        mode: 'expoGenerateApk',
+        expoSnackId,
+        iconUri,
+        splashImageUri
+      });
+      this.setState({
+        generatingApk: false,
+        apkError: null,
+        apkUri
+      });
+    } catch (e) {
+      this.setState({
+        generatingApk: false,
+        apkError: 'Failed to create Android app. Please try again later.'
+      });
+    }
+  }
+
+  onActionButton = () => {
+    const {screen} = this.state;
+
+    switch (screen) {
+      case 'intro':
+        return this.setState({screen: 'platform'});
+      // case 'export':
+      //   return this.setState({screen: 'platform'});
+      case 'platform':
+        return this.setState({screen: 'publish'});
+      // return this.setState({screen: 'icon'});
+      // case 'icon':
+      //   return this.setState({screen: 'publish'});
+      case 'publish':
+        this.publishAndGenerateApk();
+        return this.setState({screen: 'generating'});
+      case 'generating':
+        return this.close();
+      default:
+        throw new Error(`ExportAllowedDialog: Unexpected screen: ${screen}`);
+    }
+  };
+
+  renderMainContent() {
+    const {screen} = this.state;
+
+    switch (screen) {
+      case 'intro':
+        return this.renderIntroPage();
+      // case 'export':
+      //   return this.renderExportPage();
+      case 'platform':
+        return this.renderPlatformPage();
+      // case 'icon':
+      //   return this.renderIconPage();
+      case 'publish':
+        return this.renderPublishPage();
+      case 'generating':
+        return this.renderGeneratingPage();
+      default:
+        throw new Error(`ExportAllowedDialog: Unexpected screen: ${screen}`);
+    }
+  }
+
+  renderIntroPage() {
+    return (
+      <div>
+        <div style={styles.section}>
+          <p style={styles.title}>
+            Code Studio can export your project as a mobile app for iOS or
+            Android
+          </p>
+        </div>
+        <div style={styles.section}>
+          <p style={styles.p}>
+            We will create a mobile app that reflects the current state of your
```

Comment:
"Exporting will create a mobile app that you can install on your phone. You can install that app on your phone and run it without opening the Code.org website. If you make changes to your app after you export, you will need to export it again."